### ISSUE TITLE
Wrong marker weights value in MarkersReference

### ISSUE THREAD

**Issue by cpizzolato** [opensim-org/opensim-core#1443]
When using the `InverseKinematicsTool` the marker weights are read from the inverse kinematics setup file. However, the values of the marker weights in the xml and in the `MarkersReference` are different.

My understanding is that `MarkersReference::_markerNames` and `MarkersReference::_weights` should be in the same order, with the marker name matching in position with the respective weight, however this seems not to happen all the time. I have a [simple test case](/cpizzolato/opensim-core/tree/example-markerweightserror) if you want to take a look. Check for testInverseKinematicsTool among the other tests.

I think the problem is in `void MarkersReference::updateInternalWeights() const`, specifically, where names and weight are associated 
```
   // Begin by assigning default weight to each. Markers that do not have a
    // weight specified in the marker_weights property use the default weight.
    _weights.assign(getNumRefs(), get_default_weight());
    // Next fill in the marker weights that were specified in the
    // marker_weights property
    int wix = -1;
    int ix = 0;
    // Build flat lists of marker weights in the same order as the marker names
    for (const std::string &name : _markerNames) {
        wix = get_marker_weights().getIndex(name, wix);
        // Associate user weights (as specified in the marker_weights property)
        // with the corresponding marker by order of marker names
        if (wix >= 0)
            _weights[ix++] = get_marker_weights()[wix].getWeight();
    }
```

the index `ix` is incremented only when a weight in the xml is found, which makes it go out of sync when a name is not found. So, `ix` should be incremented at every iteration of the for loop.

Please, let me know it this makes sense :)

**Comment by jenhicks** [comment]
@aseth1 will look into this. Thank you for reporting Claudio!

**Comment by cpizzolato** [comment]
Hi Jen, no worries! 

To add some more details, I have a IK setup file where all the weights are 1, except for the the Top.Head marker, which has a weight of 10, see [file](https://github.com/cpizzolato/opensim-core/blob/example-markerweightserror/OpenSim/Tools/Test/subject01_IK_setup.xml).

If I print the content of `_markerNames` and `_weights` just before exiting the function  `MarkersReference::updateInternalWeights`
```
    for (int i(0); i < _markerNames.size(); ++i) {
        std::cout << _markerNames[i] << ": " << _weights[i] << std::endl;
    }
```
The marker weights should be in the same order as the marker names, however, I obtain the following output:

> R.ASIS: 1
> L.ASIS: 1
> V.Sacral: 1
> R.Thigh.Upper: 1
> R.Thigh.Front: 1
> R.Thigh.Rear: 1
> L.Thigh.Upper: 1
> L.Thigh.Front: 1
> L.Thigh.Rear: 1
> R.Shank.Upper: 1
> R.Shank.Front: 1
> R.Shank.Rear: 1
> L.Shank.Upper: 1
> L.Shank.Front: 1
> L.Shank.Rear: 1
> R.Heel: 1
> R.Midfoot.Sup: 1
> R.Midfoot.Lat: 1
> R.Toe.Tip: 1
> L.Heel: 1
> L.Midfoot.Sup: 1
> L.Midfoot.Lat: 1
> L.Toe.Tip: 1
> Sternum: 1
> R.Acromium: 1
> L.Acromium: 1
> R.Bicep: 1
> L.Bicep: 1
> R.Elbow: 1
> L.Elbow: 1
> **R.Wrist.Med: 10**
> R.Wrist.Lat: 1
> L.Wrist.Med: 1
> L.Wrist.Lat: 1
> R.Toe.Lat: 1
> R.Toe.Med: 1
> L.Toe.Lat: 1
> L.Toe.Med: 1
> R.Temple: 1
> L.Temple: 1
> **Top.Head: 1**

Where Top.Head has a weight of 1, and R.Wrist.Med has a weight of 10, which is different from what is specified in the xml.

Cheers
Claudio

**Comment by aseth1** [comment]
@cpizzolato thank you reporting this issue. I could not find `testInverseKinematicsTool` in your fork.
Did you push it to your repo?

**Comment by cpizzolato** [comment]
Hi @aseth1, the "test" is in [OpenSim/Tools/Test](https://github.com/cpizzolato/opensim-core/blob/example-markerweightserror/OpenSim/Tools/Test/testInverseKinematicsTool.cpp). I have added a couple of comments as well. Just for clarity is not a proper test, I have added a `cout` and a specific setup file, definitely nothing fancy, but hopefully shows the problem.

If you believe it's worth, I can come up with a proper test suite, so we can verify multiple conditions.
